src/checkout.ts

```diff
@@ -0,0 +1,149 @@
+import * as core from '@actions/core';
+import * as coreCommand from '@actions/core/lib/command';
+import * as fsHelper from './fs-helper';
+import * as gitSourceProvider from './git-source-provider';
+import * as github from '@actions/github'
+import * as path from 'path';
+
+async function run() {
+    try {
```

Comment:
add comment here about using error to break out of the method since there are methods below with side effects like error conditions that we expect to happen and stop continuation (and that is fully okay)

Reply:
i dont understand what the concern is?

note, i renamed the file to main.ts so it should be more clear this is the outer try/catch